### ISSUE TITLE
DeepSpeed Inference Kernel Execution Error on Tesla K80

### ISSUE THREAD

**Issue by blueviggen** [microsoft/DeepSpeed#1301]
On an NVIDIA Tesla K80 GPU VM, when I execute the following Python script with the DeepSpeed Inference API
 and a pre-trained BERT model (downloaded via HuggingFace Transformers API) , I'm getting the following kernel execution error with error code 8.

    !!!! kernel execution error. (m: 2304, n: 10, k: 768, error: 8)
    !!!! kernel execution error. (batch: 12, m: 10, n: 10, k: 64, error: 8)
    !!!! kernel execution error. (batch: 12, m: 64, n: 10, k: 10, error: 8)
    !!!! kernel execution error. (m: 768, n: 10, k: 768, error: 8)
    !!!! kernel execution error. (m: 3072, n: 10, k: 768, error: 8)
    !!!! kernel execution error. (m: 768, n: 10, k: 3072, error: 8)

In addition, the inference results are totally inconsistent with results without DeepSpeed.

@RezaYazdaniAminabadi has reported that he doesn't see this issue on NVIDIA V100. So it's quite likely this issue is specific to K80 and other generations of NVIDIA GPU hardware.

-----------------------------------------------------------------------------------------

from transformers import pipeline
import deepspeed

p = pipeline('fill-mask', model = 'bert-base-cased', device = 0)

p.model = deepspeed.init_inference(p.model, mp_size = 1,dtype = torch.float)

result = p("Hello I'm a [MASK] model.", do_sample=True, min_length=50)

**Comment by hyunwoongko** [comment]
DeepSpees kernel doesnt support k80.

**Comment by blueviggen** [comment]
@jeffra just added the following statement on GPU requirements for DeepSpeed

> Specific GPUs we develop and test against are listed below, this doesn't mean your GPU will not work if it doesn't fall into this category it's just DeepSpeed is most well tested on the following:
> **NVIDIA: Pascal, Volta, and Ampere architectures
> AMD: MI100 and MI200**

#2143
